src/GitLink/Linker.cs

```diff
@@ -36,167 +36,176 @@ public static bool Link(string pdbPath, LinkOptions options = default(LinkOption
-                        {
-                            Log.Debug("Verifying pdb file");
-
-                            var missingFiles = pdb.FindMissingOrChangedSourceFiles();
-                            foreach (var missingFile in missingFiles)
-                            {
-                                Log.Warning($"File \"{missingFile}\" missing or changed since the PDB was compiled.");
-                            }
-                        }
+                        Log.Error("Couldn't auto detect git repo. Please use -baseDir to manually set it.");
+                        return false;
                     }
                 }
 
-                string workingDirectory = Path.GetDirectoryName(repositoryDirectory);
+                sourceFiles = GetSourceFilesFromDepot(repositoryDirectory);
+            }
+            else
+            {
+                sourceFiles = GetSourceFilesFromPdb(pdbPath);
+
+                string someSourceFile = sourceFiles.FirstOrDefault();
+                if (someSourceFile == null)
+                {
+                    Log.Error("No source files were found in the PDB. If you're PDB is a native one you should use -a option.");
+                    return false;
+                }
 
-                var repository = new Lazy<Repository>(() => new Repository(repositoryDirectory));
-                try
+                if (repositoryDirectory == null)
                 {
-                    string commitId = options.CommitId ?? repository.Value.Head.Commits.FirstOrDefault()?.Sha;
-                    if (commitId == null)
+                    repositoryDirectory = GitDirFinder.TreeWalkForGitDir(Path.GetDirectoryName(sourceFiles.FirstOrDefault()));
+                    if (repositoryDirectory == null)
                     {
-                        Log.Error("No commit is checked out to HEAD. Have you committed yet?");
+                        Log.Error("No source files found that are tracked in a git repo.");
                         return false;
                     }
+                }
+            }
 
-                    var providerManager = new Providers.ProviderManager();
-                    Providers.IProvider provider;
-                    if (options.GitRemoteUrl == null)
-                    {
-                        var candidateProviders = from remote in repository.Value.Network.Remotes
-                                                 let p = providerManager.GetProvider(remote.Url)
-                                                 where p != null
-                                                 select p;
-                        provider = candidateProviders.FirstOrDefault();
-                    }
-                    else
-                    {
-                        provider = providerManager.GetProvider(options.GitRemoteUrl.AbsoluteUri);
-                    }
+            string workingDirectory = Path.GetDirectoryName(repositoryDirectory);
```

Comment:
Not supposed to. Maybe it got lost during the refactoring. I'll check tonight at home.